lib/backend/artifactorybackend/blobclient_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2016-2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package artifactorybackend
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/pressly/chi"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/kraken/core"
+	"github.com/uber/kraken/lib/backend/backenderrors"
+	"github.com/uber/kraken/utils/memsize"
+	"github.com/uber/kraken/utils/randutil"
+	"github.com/uber/kraken/utils/testutil"
+)
+
+func TestClientFactory(t *testing.T) {
+	require := require.New(t)
+
+	config := Config{}
+	f := blobClientFactory{}
+	_, err := f.Create(config, nil)
+	require.NoError(err)
+}
+
+func TestBlobDownloadBlobSuccess(t *testing.T) {
+	require := require.New(t)
+
+	blob := randutil.Blob(32 * memsize.KB)
+	namespace := core.NamespaceFixture()
+
+	r := chi.NewRouter()
+	r.Get(fmt.Sprintf("/v2/%s/blobs/:blob", namespace), func(w http.ResponseWriter, req *http.Request) {
+		_, err := io.Copy(w, bytes.NewReader(blob))
+		require.NoError(err)
+	})
+	r.Head(fmt.Sprintf("/v2/%s/blobs/:blob", namespace), func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Set("Content-Length", fmt.Sprintf("%d", len(blob)))
+	})
```

Comment:
could you scope these into a function? so it can be shared by all tests in this file.